pkgs/by-name/ps/pspy/package.nix

```diff
@@ -0,0 +1,99 @@
+{
+  buildGoModule,
+  fetchFromGitHub,
+  lib,
+  upx,
+  versionCheckHook,
+  stdenv,
+}:
+let
+  inherit (stdenv.hostPlatform) system;
+  arch =
+    {
+      aarch64-linux = {
+        _32 = "arm";
+        _64 = "arm64";
+      };
+
+      x86_64-linux = {
+        _32 = "386";
+        _64 = "amd64";
+      };
+    }
+    .${system} or (throw "Unsupported system ${system}!");
+in
+buildGoModule (finalAttrs: {
+  pname = "pspy";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "DominicBreuker";
+    repo = "pspy";
+    tag = "v${finalAttrs.version}";
+
+    leaveDotGit = true;
+    postFetch = ''
+      git -C $out rev-parse HEAD > $out/.git_head
+      rm -rf $out/.git
+    '';
+
+    hash = "sha256-CPWoKxmjlGYP2kAC+LscOtrPpUjzpRoGTeohlw0mmh4=";
+  };
+
+  vendorHash = "sha256-mgAsy2ufMDNpeCXG/cZ10zdmzFoGfcpCzPWIABnvJWU=";
+
+  env.CGO_ENABLED = "0";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${finalAttrs.version}"
+  ];
+
+  nativeBuildInputs = [ upx ];
+
+  # we build both 32 and 64 bit binaries just like upstream does
+  # however our "small" versions are not dynamically linked, only compressed
+  # since dynamically linked binaries would refer to missing Nix store paths
+  # when they are copied onto a target system
+  buildPhase = ''
+    runHook preBuild
+
+    ldflags+=" -X main.commit=$(<.git_head)"
+
+    GOARCH=${arch._32} go build -ldflags "$ldflags" -o $out/bin/pspy32
+    GOARCH=${arch._64} go build -ldflags "$ldflags" -o $out/bin/pspy64
+
+    upx $out/bin/pspy32 -o $out/bin/pspy32s
+    upx $out/bin/pspy64 -o $out/bin/pspy64s
+
+    runHook postBuild
+  '';
+
+  # the various TestStart* tests defined in $src/internal/pspy/pspy_test.go
+  # can in rare cases hit a race condition
+  # ("Did not get message in time" or "Wrong message")
+  checkPhase = ''
+    runHook preCheck
+
+    go test -skip='^TestStart' ./...
+
+    runHook postCheck
+  '';
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--help";
+  doInstallCheck = true;
+
+  meta = {
+    description = "Monitor linux processes without root permissions";
+    homepage = "https://github.com/DominicBreuker/pspy";
+    license = lib.licenses.gpl3;
```

Comment:
Only or Plus?

Reply:
@42LoCo42 According to https://github.com/DominicBreuker/pspy?tab=GPL-3.0-1-ov-file#readme I believe it is `gpl3Only` and not `gpl3Plus`